src/panel/TemplateEditor/lib/cloud/user.ts

```diff
@@ -2,16 +2,37 @@
 import {
     getFirestore,
     onSnapshot,
-    doc
+    doc,
+    Timestamp
 } from "firebase/firestore";
 import { initializeApp } from "firebase/app";
 import * as request from "./request";
 
+function formatDocument(obj: unknown): unknown {
```

Comment:
the cause of some pretty small / hard-to-diagnose issues... the validation of the user... we might move that 100% to servers-side, to reduce this redundancy?